index.js

```diff
@@ -0,0 +1,57 @@
+const AWS = require('aws-sdk');
+
+/**
+ * @param {object} arc - the parsed .arc file currently executing
+ * @param {object} cloudformation - the current AWS::Serverless CloudFormation template
+ * @param {object} stage - the application stage (one of `staging` or `production`)
+ */
+module.exports = async function arcBucketMacro(arc, cloudformation, stage) {
+  const config = arc.s3;
+  const buckets = {};
+  const aws = new AWS.S3();
+  // for each bucket:
+  await Promise.all(config.map(b => {
```

Comment:
lets only create the stage bucket if this is deploying the stage environment, same with production.  might simplify this code

Reply:
This should already skip any bucket config that does not match 'stage'.

Reply:
why just stage?